> This notebook can be used to generate mutant sequences

In [2]:
import pandas as pd
import numpy as np 

### 1. Single Mutant sequences

In [12]:
def generate_mutations(wt_seq):
    mutations = []
    for i in range(len(wt_seq)):
        for amino_acid in 'ACDEFGHIKLMNPQRSTVWY':
            if amino_acid != wt_seq[i]:
                mutated_seq = list(wt_seq)
                mutated_seq[i] = amino_acid
                mutated_seq = ''.join(mutated_seq)

                mutation = {
                    'Position': i + 1,
                    'Original_AA': wt_seq[i],
                    'Mutated_Position': i + 1,
                    'Mutated_AA': amino_acid,
                    'seq': mutated_seq
                }
                mutations.append(mutation)
    return mutations

if __name__ == "__main__":
    wt_seq = 'MLDMGQDRPIDGSGAPGADDTRVEVQPPAQWVLDLIEASPIASVVSDPRLADNPLIAINQAFTDLTGYSEEECVGRNCRFLAGSGTEPWLTDKIRQGVREHKPVLVEILNYKKDGTPFRNAVLVAPIYDDDDELLYFLGSQVEVDDDQPNMGMARRERAAEMLKTLSPRQLEVTTLVASGLRNKEVAARLGLSEKTVKMHRGLVMEKLNLKTSADLVRIAVEAGI'
    mutations = generate_mutations(wt_seq)
    
    # Create a DataFrame from the mutations list
    df = pd.DataFrame(mutations)

    # Reorder the columns
    df = df[['Position', 'Original_AA', 'Mutated_Position', 'Mutated_AA', 'seq']]

    # Save the DataFrame to a CSV file
    df.to_csv('single_mutants.csv', index=False)


##### Remarks:

In [10]:
print(19*len(wt_seq))

4275


> There are 4275 possible sequences for single mutant proteins.

> 